File: shared/src/commonMain/kotlin/com/wellnesswingman/data/model/TrackedEntry.kt
package com.wellnesswingman.data.model

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

/**
 * Represents a tracked health entry (meal, exercise, sleep, etc.).
 */
@Serializable
data class TrackedEntry(
    val entryId: Long = 0,
    val externalId: String? = null,
    val entryType: EntryType = EntryType.UNKNOWN,
    val capturedAt: Instant,
    val capturedAtTimeZoneId: String? = null,
    val capturedAtOffsetMinutes: Int? = null,
    val blobPath: String? = null,
    val dataPayload: String = "",
    val dataSchemaVersion: Int = 1,
    val processingStatus: ProcessingStatus = ProcessingStatus.PENDING,

    /**
     * User-provided notes (text or voice transcription) captured at time of photo submission.
     * This field persists independently of LLM analysis and remains available for corrections.
     */
    val userNotes: String? = null
) {
    /**
     * Returns the local captured time using timezone information.
     * Note: For full implementation, this would need platform-specific timezone handling.
     */
    fun getCapturedAtLocal(): Instant {
        // For now, just return the UTC time
        // TODO: Implement proper timezone conversion using platform-specific APIs
        return capturedAt
    }
}

File: archive/WellnessWingman/Models/TrackedEntry.cs

using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using WellnessWingman.Utilities;

namespace WellnessWingman.Models;

public class TrackedEntry
{
    public int EntryId { get; set; }
    public Guid? ExternalId { get; set; }
    public EntryType EntryType { get; set; } = EntryType.Unknown;
    public DateTime CapturedAt { get; set; }
    public string? CapturedAtTimeZoneId { get; set; }
    public int? CapturedAtOffsetMinutes { get; set; }
    public string? BlobPath { get; set; }
    public string DataPayload { get; set; } = string.Empty;
    public int DataSchemaVersion { get; set; }
    public ProcessingStatus ProcessingStatus { get; set; } = ProcessingStatus.Pending;

    /// <summary>
    /// User-provided notes (text or voice transcription) captured at time of photo submission.
    /// This field persists independently of LLM analysis and remains available for corrections.
    /// </summary>
    public string? UserNotes { get; set; }

    [NotMapped]
    [JsonIgnore]
    public IEntryPayload Payload { get; set; } = new PendingEntryPayload();

    [NotMapped]
    [JsonIgnore]
    public DateTime CapturedAtLocal => DateTimeConverter.ToOriginalLocal(
        CapturedAt,
        CapturedAtTimeZoneId,
        CapturedAtOffsetMinutes);
}
